misk-hibernate/src/main/kotlin/misk/hibernate/Destination.kt

```diff
@@ -0,0 +1,61 @@
+package misk.hibernate
+
+data class Destination(
+  val keyspace: Keyspace?,
+  val shard: Shard?,
+  val tabletType: TabletType?
+) {
+  init {
+    if (shard != null) {
+      check(keyspace == shard.keyspace)
+    }
+  }
+
+  constructor(shard: Shard) : this(shard.keyspace, shard, null)
+
+  constructor(tabletType: TabletType) : this(null, null, tabletType)
+
+  override fun toString(): String {
+    val tabletType = if (tabletType != null) "@$tabletType" else ""
```

Comment:
nit: should check if tabletType already starts with '@'?

Reply:
I don't think it can. It's an enum! :-)

Reply:
Makes sense! :-)